orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -129,7 +157,31 @@ class BakeStage implements StageDefinitionBuilder {
           return deploymentDetails
         }
       ]
+
+      if (failOnImageNameMismatchEnabled() && !stage.context.dontFailOnImageNameMismatch) {
+         List<String> distinctImageNames = globalContext.deploymentDetails.stream()
+          .map({details -> details['imageName']})
+          .distinct()
+          .collect(Collectors.toList())
+
+        Joiner joiner = Joiner.on(", ")
```

Comment:
Why create this object here if it is only used within the following conditional?